ludwig/explain/captum.py

```diff
@@ -215,6 +187,80 @@ def explain(self) -> Tuple[List[Explanation], List[float]]:
         return self.explanations, expected_values
 
 
+def get_input_tensors(
+    model: LudwigModel, input_set: pd.DataFrame, run_config: ExplanationRunConfig
+) -> List[torch.Tensor]:
+    """Convert the input data into a list of variables, one for each input feature.
+
+    # Inputs
+
+    :param model: The LudwigModel to use for encoding.
+    :param input_set: The input data to encode of shape [batch size, num input features].
+
+    # Return
+
+    :return: A list of variables, one for each input feature. Shape of each variable is [batch size, embedding size].
+    """
+    # Ignore sample_ratio from the model config, since we want to explain all the data.
+    sample_ratio_bak = model.config_obj.preprocessing.sample_ratio
+    model.config_obj.preprocessing.sample_ratio = 1.0
+
+    config = model.config_obj.to_dict()
+    training_set_metadata = copy.deepcopy(model.training_set_metadata)
+    for feature in config[INPUT_FEATURES]:
+        preprocessing = training_set_metadata[feature[NAME]][PREPROCESSING]
+        if preprocessing.get("cache_encoder_embeddings"):
+            preprocessing["cache_encoder_embeddings"] = False
+
+    # Convert raw input data into preprocessed tensor data
+    dataset, _ = preprocess_for_prediction(
+        config,
+        dataset=input_set,
+        training_set_metadata=training_set_metadata,
+        data_format="auto",
+        split="full",
+        include_outputs=False,
+        backend=model.backend,
+        callbacks=model.callbacks,
+    )
+
+    # Restore sample_ratio
+    model.config_obj.preprocessing.sample_ratio = sample_ratio_bak
+
+    # Make sure the number of rows in the preprocessed dataset matches the number of rows in the input data
+    assert (
+        dataset.to_df().shape[0] == input_set.shape[0]
+    ), f"Expected {input_set.shape[0]} rows in preprocessed dataset, but got {dataset.to_df().shape[0]}"
+
+    # Convert dataset into a dict of tensors, and split each tensor into batches to control GPU memory usage
+    inputs = {
+        name: torch.from_numpy(dataset.dataset[feature.proc_column]).split(run_config.batch_size)
+        for name, feature in model.model.input_features.items()
+    }
+
+    # Dict of lists to list of dicts
+    input_batches = [dict(zip(inputs, t)) for t in zip(*inputs.values())]
+
+    # List of dicts to dict of lists
+    preproc_inputs = {k: torch.cat([d[k] for d in input_batches]) for k in input_batches[0]}
+
+    data_to_predict = [v for _, v in preproc_inputs.items()]
+    tensors = []
+    for t in data_to_predict:
+        if t.dtype == torch.int8 or t.dtype == torch.int16 or t.dtype == torch.int32 or t.dtype == torch.int64:
```

Comment:
Could this be simplified to `if not torch.is_floating_point(t.dtype)`? The body of the block could then be reorganized to handle the bool case.

Reply:
Good point, I can leave a TODO for now.